fix(migrate): use the right node version (without >=)

clementprevot opened the pull request.

Commit:
fix(migrate): use the right node version (without >=)

clementprevot commented:
My previous fix for this was bugged because of a mis-interpretation of the node version to a redirect.

Kikobeats commented:
@clementprevot I refactor the code. Can you merge master?

clementprevot commented:
I saw that, I was merging the conflicts. You are way too fast for me :wink:

Commit:
fix(migrate): rename the `rangeVersion` arg to `currentVersion`

Kikobeats commented:
haha, but you resolve the conflict using the previous version!

clementprevot commented:
Really? Hum that's weird, let me check that

clementprevot commented:
@Kikobeats Are you sure? When I look at the changes of the PR, I only see the changes to use the right node version (variable name `currentVersion` in both `index.js` and `switcher.js` files)

Kikobeats commented on bin/index.js:
```diff
@@ -27,8 +27,10 @@ var rangeVersion = pkg.engines && pkg.engines.node
 
 config(function (versions) {
+  var currentVersion = process.versions.node
+
```
can you remove this extra line?

clementprevot replied:
Done

Kikobeats commented on bin/index.js:
```diff
@@ -27,8 +27,10 @@ var rangeVersion = pkg.engines && pkg.engines.node
 if (!rangeVersion) processExit()
 
 config(function (versions) {
+  var currentVersion = process.versions.node
```
If you are tying to switch the version, this version refers the old version.

For sanetize the version, use `semver.sanetize`:

```
semver.clean('  =v1.2.3   ') // '1.2.3'
```

clementprevot replied:
No, I precisely want to get the current (so the old) version of node, to be able to tell to NVM to migrate already installed package from this version.

e.g: we are in node 6.7.0, NVM detects a new 6.8.1 version, I want to migrate packages from 6.7.0 so the command line is `nvm install 6.8.1 --resinstall-packages-from=6.7.0`. That's the main goal of the `currentVersion` param given to the `createSwitcher` function

Kikobeats commented:
@clementprevot lol you're right, I added comments 👍

Kikobeats commented on bin/switcher.js:
```diff
@@ -2,20 +2,20 @@
 
 var spawn = require('child_process').spawn
 
```
`var semver = require('semver')` here

clementprevot replied:
See my previous comment, thus, not needed

Kikobeats commented on bin/switcher.js:
```diff
@@ -2,20 +2,20 @@
+function nvmCommand (maxSatisfyVersion, currentVersion) {
   var cmd = ''
   cmd += 'nvm install '
   cmd += maxSatisfyVersion
```
`cmd += semver.clean(maxSatisfyVersion)` here

clementprevot replied:
See again my first comment

clementprevot committed:
fix(migrate): remove extra-line

Commit:
feat(migrate): don't reinstall if same version

NVM still reinstall package from even if no new node version has been
installed

Kikobeats commented:
Need to resolve the pending reviews before merge this

clementprevot commented:
I don't see any pending review

Kikobeats commented:
was (again) my fault, check now 😄

clementprevot commented:
I commented on your reviews. I think you misunderstood what I wanted to do :wink:

Kikobeats commented:
yay!
